### PR TITLE
Add project references

### PR DESCRIPTION
**Pull request by tnorling**
Adds TypeScript [Project References](https://www.typescriptlang.org/docs/handbook/project-references.html) to the 2.x libraries. 

This fixes VS Code issues with "Go to definition" and "Find all references" when working across multiple linked libraries e.g. msal-common and msal-browser

### CODE REVIEW

File: lib/msal-browser/tsconfig.json [hunk 1]
```diff
@@ -15,8 +15,14 @@
...
     },
+    "references": [
+        {
+            "path": "../msal-common"
+        }
+    ],
     "include": [
-        "src/**/*"
+        "src/**/*",
+        "./package.json"
```
Comment (tnorling) [inline lib/msal-browser/tsconfig.json:25]: Temporarily needed until #2915 is merged

File: lib/msal-common/tsconfig.json [hunk 2]
```diff
@@ -26,7 +26,8 @@
         "noFallthroughCasesInSwitch": true,
     },
     "include": [
-        "src/**/*"
+        "src/**/*",
+        "./package.json"
```
Comment (tnorling) [inline lib/msal-common/tsconfig.json:30]: Temporarily needed until #2915 is merged

**Review by jasonnutter — APPROVED** [review]
Big fan!

**Comment by jasonnutter** [comment]
Also, whats up the msal react package-lock diff?

**Comment by tnorling** [comment]
> Also, whats up the msal react package-lock diff?

Indents got all messed up. Reverted.

**Review by pkanher617 — APPROVED** [review]

**Review by hectormmg — APPROVED** [review]

**Pull request merged**